src/Property.php

```diff
@@ -143,6 +143,41 @@ protected function cast($value)
         return new $castTo($value);
     }
 
+    protected function castArray(array $values)
+    {
+        foreach ($values as $value) {
```

Comment:
Can you explain why this loop is here? It feels like it should be done on a higher level, maybe together with the `Arr::isAssoc($value)` check?

Reply:
Thinking about it, perhaps we shouldn't use `Arr::isAssoc($value)` but instead a separate function called something like "shouldBeCastToCollection", which would loop once over all items, and check whether

- The keys are no strings, to replace `isAssoc`
- The values are arrays

Reply:
So, initially the goal of that loop was to validate that the array is an array of arrays. I agree with your point though that this validation should be separated out. Renaming it, especially in context with your further comments, also helps contribute to the consistency of the naming within the package; I'll make sure to include that in my next update.